admin/src/js/services/message-queue.js

```diff
@@ -109,32 +107,35 @@ angular.module('services').factory('MessageQueue',
         });
     };
 
-    const getPatientsAndRegistrations = function(messages, settings) {
-      const patientIds = compactUnique(messages.map(function(message) {
-        // don't process items which already have generated messages
-        return !message.sms && message.context.patient_id;
-      }));
+    const getSubjectsAndRegistrations = (messages, settings) => {
+      const shortcodes = compactUnique(messages.reduce((shortcodes, message) => {
```

Comment:
@garethbowen The issues I've seen with `reduce` over alternatives has been around increased complexity (to a future reader trying to think through what's going on) and concerns around object creation. For the benefit of developers doing this going forward, would you please add the guidance and rationale to the coding standards docs? An eslint rule would be nice too, but I don't know if this can be accurately be checked there.